plugins/chromatic_correction.py

```diff
@@ -0,0 +1,355 @@
+                da.metadata.update(transformation_dict[model.MD_CHROMATIC_COR][band_pos])
+
+            # Extra information saved for tracking/debugging purpose
+            input_wl = da.metadata[model.MD_IN_WL]
+            output_wl = da.metadata[model.MD_OUT_WL]
+            transformation_dict["Extra Information"][band_pos] = {
+                "tint": list(da.metadata[model.MD_USER_TINT]),
+                "excitation": [float(input_wl[0]), float(input_wl[1])],
+                "emission": [float(output_wl[0]), float(output_wl[1])],
+                "exposure time": float(da.metadata[model.MD_EXP_TIME]),
+                "light power": float(da.metadata[model.MD_LIGHT_POWER])
+            }
+
+            # Save the modified image and display
+            file_location = get_file_location(f"{da.metadata[model.MD_DESCRIPTION]}_modified_metadata",
+                                              "ome.tiff")
+            self.display_static_stream(file_location, da)
+
+        return transformation_dict
+
+    def open_dialog_window(self):
+        """Opens a dialog window which shows start button if all the specific requirements are present
+        otherwise suggests changing the settings"""
+        localization_tab = self.main_app.main_data.getTabByName("cryosecom-localization")
+        tab_data = localization_tab.tab_data_model
+        all_emission = [numpy.array(fms.emission.value) for fms in
+                        tab_data.streams.value]
+        # reference channel should be present in the present streams
+        check_ref_channel = any(
+            all(numpy.isclose(emission, self.ref_emission, rtol=1e-09, atol=1e-09)) for emission in
+            all_emission)
+        # No two streams should have same emission
+        unique_elements = set(map(tuple, all_emission))
+        check_unique_channels = len(unique_elements) == len(all_emission)
+        self.dlg = AcquisitionDialog(self, "Chromatic Correction",
+                                     ("Ready to start Chromatic aberration on the given channel.\n"
+                                      "The process will take about 5 minutes\n\n"
+                                      "Z stack range is {} m\n"
+                                      "Z step size of {} m".format(Z_RANGE, Z_STEP_SIZE)))
+        if (len(tab_data.streams.value) <= 1) or (not check_ref_channel) or (not check_unique_channels):
+            txt = ("Problems\n1. Add minimum of two streams.\n2. One of the streams must be green channel.\n"
+                   "3. No two streams should be the same.")
+            self.dlg.lbl_description.SetLabel(txt)
+            self.dlg.addButton("Manually fix")
+            self.dlg.ShowModal()
+            return
+        self.dlg.Bind(wx.EVT_CLOSE, self.on_close)
+        self.dlg.addButton("Start", self._on_start, face_colour='blue')
+        self.dlg.ShowModal()
+
+    def on_close(self, evt, *args, **kwargs):
```

Comment:
Why `*args, **kwargs`? I don't think they are needed. It seems the caller either is a event binder (only passes `evt`), or from `_on_start()` (no real argument at all).